commit 8540bd1b990bad7f7e95b5bf1adf30bfaf2e38c9
Merge: 3d5689e01a42 3b467b16582c
Author: Marc Zyngier <maz@kernel.org>
Date:   Fri May 3 11:39:52 2024 +0100

    Merge branch kvm-arm64/pkvm-6.10 into kvmarm-master/next
    
    * kvm-arm64/pkvm-6.10: (25 commits)
      : .
      : At last, a bunch of pKVM patches, courtesy of Fuad Tabba.
      : From the cover letter:
      :
      : "This series is a bit of a bombay-mix of patches we've been
      : carrying. There's no one overarching theme, but they do improve
      : the code by fixing existing bugs in pKVM, refactoring code to
      : make it more readable and easier to re-use for pKVM, or adding
      : functionality to the existing pKVM code upstream."
      : .
      KVM: arm64: Force injection of a data abort on NISV MMIO exit
      KVM: arm64: Restrict supported capabilities for protected VMs
      KVM: arm64: Refactor setting the return value in kvm_vm_ioctl_enable_cap()
      KVM: arm64: Document the KVM/arm64-specific calls in hypercalls.rst
      KVM: arm64: Rename firmware pseudo-register documentation file
      KVM: arm64: Reformat/beautify PTP hypercall documentation
      KVM: arm64: Clarify rationale for ZCR_EL1 value restored on guest exit
      KVM: arm64: Introduce and use predicates that check for protected VMs
      KVM: arm64: Add is_pkvm_initialized() helper
      KVM: arm64: Simplify vgic-v3 hypercalls
      KVM: arm64: Move setting the page as dirty out of the critical section
      KVM: arm64: Change kvm_handle_mmio_return() return polarity
      KVM: arm64: Fix comment for __pkvm_vcpu_init_traps()
      KVM: arm64: Prevent kmemleak from accessing .hyp.data
      KVM: arm64: Do not map the host fpsimd state to hyp in pKVM
      KVM: arm64: Rename __tlb_switch_to_{guest,host}() in VHE
      KVM: arm64: Support TLB invalidation in guest context
      KVM: arm64: Avoid BBM when changing only s/w bits in Stage-2 PTE
      KVM: arm64: Check for PTE validity when checking for executable/cacheable
      KVM: arm64: Avoid BUG-ing from the host abort path
      ...
    
    Signed-off-by: Marc Zyngier <maz@kernel.org>

diff --cc arch/arm64/kvm/hyp/nvhe/tlb.c
index 2fc68da4036d,c6e0a49eb860..ca3c09df8d7c
--- a/arch/arm64/kvm/hyp/nvhe/tlb.c
+++ b/arch/arm64/kvm/hyp/nvhe/tlb.c
@@@ -152,10 -219,9 +219,10 @@@ void __kvm_tlb_flush_vmid_range(struct 
  	start = round_down(start, stride);
  
  	/* Switch to requested VMID */
- 	__tlb_switch_to_guest(mmu, &cxt, false);
+ 	enter_vmid_context(mmu, &cxt, false);
  
 -	__flush_s2_tlb_range_op(ipas2e1is, start, pages, stride, 0);
 +	__flush_s2_tlb_range_op(ipas2e1is, start, pages, stride,
 +				TLBI_TTL_UNKNOWN);
  
  	dsb(ish);
  	__tlbi(vmalle1is);